asf_heat_pump_suitability/pipeline/prepare_features/anchor_properties.py

```diff
@@ -0,0 +1,170 @@
+    "sports_and_recreation_venue",
+    "water_park",
+    "ice_skating_rink",
+    "swimming_pool",
+    "jail_and_prison",
+    "shopping_center",
+    "airport_terminal",
+    "airport",
+    "superstore",
+    "department_store",
+}
+
+
+def _safe_join_categories(categories: pd.Series) -> str:
+    """
+    Safely join category strings.
+
+    Args:
+        categories: Series of categories to join
+
+    Returns:
+        str: Comma-separated string of unique categories
+    """
+    valid_categories = set(str(cat) for cat in categories if pd.notna(cat))
+    return ", ".join(sorted(valid_categories)) if valid_categories else ""
+
+
+def load_and_process_poi() -> gpd.GeoDataFrame:
+    """
+    Load and process Points of Interest data.
+
+    Returns:
+        GeoDataFrame: Processed POI data containing only anchor properties
+
+    Raises:
+        ValueError: If required columns are missing
+    """
+    logger.info("Loading POI data...")
+    poi = load_s3_data("asf-heat-pump-suitability", "source_data/poi_uk.gpkg")
+
+    # Validate POI data
+    required_columns = {"country", "main_category", "geometry"}
+    missing_columns = required_columns - set(poi.columns)
+    if missing_columns:
+        raise ValueError(f"Missing required columns in POI data: {missing_columns}")
+
+    # Filter and process
+    poi = poi[poi.country == '"GB"'].copy()
+    poi["main_category"] = poi["main_category"].apply(
+        lambda x: str.strip(x, '"') if pd.notna(x) else None
+    )
+
+    # Filter anchor properties and reproject
+    anchor_properties = (
+        poi[poi.main_category.isin(ANCHOR_PROPERTIES)]
+        .to_crs(INPUT_CRS)
+        .to_crs(PROCESSING_CRS)
+    )
+
+    logger.info(f"Found {len(anchor_properties)} potential anchor properties")
+    return anchor_properties
+
+
+def identify_anchor_properties() -> gpd.GeoDataFrame:
+    """
+    Identify and analyze anchor properties within LSOAs.
+
+    Returns:
+        GeoDataFrame: Summary of anchor properties by LSOA containing columns:
+            - lsoa: Unique identifier for the LSOA
+            - lsoa_name: Name of the LSOA
+            - anchor_count: Number of anchor properties in the LSOA
+            - building_categories: List of main categories present
+            - building_subcategories: List of subcategories present
+            - has_anchor_property: Boolean indicating presence of anchor properties
+    """
+    try:
+        logger.info("Starting anchor property analysis...")
+
+        # Load required data
+        lsoa_gdf = logger.info("Loading LSOA boundaries...")
+        lsoa_gdf = gpd.read_file(
+            "s3://asf-heat-pump-suitability/source_data/Lower_layer_Super_Output_Areas_2021_EW_BFE_V9_-9107090204806789093/LSOA_2021_EW_BFE_V9.shp"
+        )
+        anchor_properties = load_and_process_poi()
+
+        lsoa_with_anchors = gpd.sjoin(
+            lsoa_gdf, anchor_properties, how="left", predicate="intersects"
+        )
```

Comment:
I think because of the `intersects` predicate we end up with more LSOA + POI pairs than there are POIs, and I can see some POIs appear multiple times in the results. I'm guessing this is POIs on the boundary of 2+ LSOAs.

I think this is something to discuss in a stand up whether we want to allow this or not because I'm not sure. What do you think?

Reply:
I think we're happy to proceed with allowing POIs on boundaries to be allocated to multiple LSOAs as from a practical standpoint, it's reasonable to assume a heat network can span multiple LSOAs.

(As discussed)